packages/ord-connect/src/lib/signMessage.ts

```diff
@@ -0,0 +1,40 @@
+import { ordit } from "@sadoprotocol/ordit-sdk";
+import { Network, Wallet } from "../providers/OrdContext.tsx";
+
+interface SignMessageParams {
+  message: string;
+  wallet: Wallet;
+  address: string | Record<string, string>;
+  network: Network;
+}
+
+export default async function getSignedMessage({
+  message,
+  wallet,
+  address,
+  network,
+}: SignMessageParams) {
+  if (wallet === Wallet.UNISAT) {
+    const signedMessage = await ordit.unisat.signMessage(message);
+    return signedMessage.base64;
+  }
+
+  if (wallet === Wallet.XVERSE) {
+    let xverseAddress = address as string;
```

Comment:
I think ternary here might be better instead of forcing types.
`const xverseAddress = typeof address === "object" ? address.ordinals : address;`

But I'm not 100% sure that TS will be satisfied with this one.

Reply:
i think the latest changes would resolve this